oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/LongUtils.java

```diff
@@ -51,4 +52,83 @@ public static long safeAdd(long a, long b) {
     public static long calculateExpirationTime(long expiration) {
         return LongUtils.safeAdd(expiration, new Date().getTime());
     }
+
+    /*
+     * Taken from https://github.com/google/guava/blob/0c33dd12b193402cdf6962d43d69743521aa2f76/guava/src/com/google/common/primitives/Longs.java#L334
+     */
+    private static final class AsciiDigits {
+        private AsciiDigits() {
+        }
+
+        private static final byte[] asciiDigits;
+
+        static {
+            byte[] result = new byte[128];
+            Arrays.fill(result, (byte) -1);
+            for (int i = 0; i < 10; i++) {
+                result['0' + i] = (byte) i;
+            }
+            for (int i = 0; i < 26; i++) {
+                result['A' + i] = (byte) (10 + i);
+                result['a' + i] = (byte) (10 + i);
+            }
+            asciiDigits = result;
+        }
+
+        static int digit(char c) {
+            return (c < 128) ? asciiDigits[c] : -1;
+        }
+    }
+
+    /**
+     * Parses the specified string as a signed decimal long value. Unlike {@link Long#parseLong(String)}, this method
+     * returns {@code null} instead of throwing an exception if parsing fails. This can be significantly more efficient
+     * when the string to be parsed is often invalid, as raising an exception is an expensive operation.
+     *<p>
+     * This is a simplified version of
```

Comment:
If we simplify, then I would also hardcode the radix to 10, and get rid of the AsciiDigits class...

Reply:
I did not look carefully at the algorithm, I just assumed that it is correct, given that it is part of Guava. So I did not want to make more changes than those that seemed totally obvious and safe. And even so, I introduced a bug when replacing the logic for checking for null/empty strings. I agree it would be nice to get rid of AsciiDigits, but I'm not sure if it is worth the effort and risk.

Reply:
I think replacing "radix" with a constant RADIX should be simple (given that the value is fixed). That would be a compromise.

Reply:
Done.

Reply:
Done, I created a constant, that's cleaner.